packages/dart_frog_test/example/test/routes/index_test.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:io';
+
+import 'package:dart_frog_test/dart_frog_test.dart';
+import 'package:test/test.dart';
+
+import '../../routes/index.dart' as route;
+
+void main() {
+  testRouteHandler(
```

Comment:
I'm not sure whether this API is more readable then the default tests personally:

```dart
import 'dart:io';

import 'package:dart_frog/dart_frog.dart';
import 'package:mocktail/mocktail.dart';
import 'package:test/test.dart';

import '../../routes/index.dart' as route;

class _MockRequestContext extends Mock implements RequestContext {}

void main() {
  group('GET /', () {
    test('responds with a 200 and "Welcome to Dart Frog!".', () {
      final context = _MockRequestContext();
      final response = route.onRequest(context);
      expect(response.statusCode, equals(HttpStatus.ok));
      expect(
        response.body(),
        completion(equals('Welcome to Dart Frog!')),
      );
    });
  });
}
```

Imo the original test is much easier to follow/understand 🤷 

I'm not sure that a test abstraction is necessary

Reply:
The snippet you provided is missing the mocking of the Request right?

That is basically the only help that the abstraction provide

Reply:
The snippet includes the mocking of the request as well `final context = _MockRequestContext();`

Reply:
It mocks the request context only, I am mentioning the Request i.e 'context.request'.

My point is that, yes I agree with you, in your example where the route handler only return a body message, that when it does not read anything from the request, the proposed API in this PR provides little to none benefit.

But when you want to test a handler that needs the request to be mocked, with headers, method, body, etc. Then it could help the user to write tests with fewer code, since the testRouteHandler will mock everything for ya, you just pass the values.

Reply:
Ah sorry I misunderstood. In that case, maybe `package:dart_frog_test` should focus on exporting classes like MockRequestContext, MockRequest, and utilities that make working with those classes easier rather than providing a custom test method?

Reply:
Yeah! That is an interesting idea, it will certainly make this package smaller